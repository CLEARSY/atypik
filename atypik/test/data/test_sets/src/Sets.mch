/* Sets
* Author: vdemeulenaere
* Creation date: 10/05/2023
*/
MACHINE
    Sets
    
SETS 
    ENS = {Elt1, Elt2};
    ABS_ENS
    
ABSTRACT_CONSTANTS 
    cst, cst2
    
ABSTRACT_VARIABLES
    ens, ens2, ens3, ens4, ens5,
    diff, union_, inter_, union_2, inter_2,
    union_3, inter_3
    
PROPERTIES
    cst : ABS_ENS &
    cst2 : BOOL
    
INVARIANT 
    ens : POW(BOOL * INTEGER) &
    ens2 : POW1(INTEGER) &
    ens3 : FIN(ens2) &
    ens4 : FIN1(ens3) &
    ens5 : POW(POW(ABS_ENS) * POW(REAL) * POW(FLOAT)) &
    diff : POW(INTEGER) &
    union_ : POW(NAT) &
    inter_ : POW(NAT1) &
    union_2 : POW(NATURAL) &
    inter_2 : POW(NATURAL1) &
    union_3 : POW(INTEGER * ABS_ENS) &
    inter_3 : POW(INTEGER * ABS_ENS)
    
INITIALISATION
    ens5 := {ens, ens2, ens3 | ens <: ABS_ENS & ens2 <: REAL & ens3 <: FLOAT} ||
    ens := {TRUE, FALSE} * (0..10) ||
    ens2 := {1} ||
    ens3 := {0} ||
    ens4 := {4} ||
    diff := ens2 - ens3 ||
    union_ := ens2 \/ ens3 ||
    inter_ := ens2 /\ ens3 ||
    union_2 := union({ens2, ens3}) ||
    inter_2 := inter({ens2, ens3}) ||
    union_3 := UNION (cst, cst2).(cst : INT & cst2 : ABS_ENS | {cst} * {cst2}) ||
    inter_3 := INTER (cst, cst2).(cst : INT & cst2 : ABS_ENS | {cst} * {cst2})

END
