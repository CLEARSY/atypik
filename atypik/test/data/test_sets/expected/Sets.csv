Expression	Identifier	Line	Column	Type
"({cst}) *s ({cst2})"	117	50	70	(POW (PRODUCT INTEGER ABS_ENS))
"cst2"	114	50	28	ABS_ENS
"({cst}) *s ({cst2})"	108	49	70	(POW (PRODUCT INTEGER ABS_ENS))
"{cst2}"	107	49	72	(POW ABS_ENS)
"{cst}"	106	49	64	(POW INTEGER)
"INT"	105	49	41	(POW INTEGER)
"cst2"	104	49	28	ABS_ENS
"inter({ens2,ens3})"	101	48	16	(POW INTEGER)
"{ens2,ens3}"	100	48	22	(POW (POW INTEGER))
"{ens2,ens3}"	97	47	22	(POW (POW INTEGER))
"cst"	103	49	23	INTEGER
"(ens2) /\ (ens3)"	96	46	20	(POW INTEGER)
"(ens2) \/ (ens3)"	95	45	20	(POW INTEGER)
"{4}"	93	43	13	(POW INTEGER)
"4"	92	43	14	INTEGER
"{0}"	91	42	13	(POW INTEGER)
"0"	90	42	14	INTEGER
"{1}"	89	41	13	(POW INTEGER)
"FIN1(ens3)"	31	28	12	(POW (POW INTEGER))
"POW1(INTEGER)"	27	26	12	(POW (POW INTEGER))
"POW((BOOL) *s (INTEGER))"	25	25	11	(POW (POW (PRODUCT BOOL INTEGER)))
"INTEGER"	21	25	22	(POW INTEGER)
"(ens2) -s (ens3)"	94	44	18	(POW INTEGER)
"BOOL"	20	25	15	(POW BOOL)
"inter_3"	17	18	14	(POW (PRODUCT INTEGER ABS_ENS))
"union_3"	16	18	5	(POW (PRODUCT INTEGER ABS_ENS))
"inter_"	13	17	19	(POW INTEGER)
"diff"	11	17	5	(POW INTEGER)
"{}"	5	-1	-1	(POW ABS_ENS)
"ENS"	0	9	5	(POW ENS)
"Elt2"	2	9	18	ENS
"POW(REAL)"	36	29	31	(POW (POW REAL))
"ABS_ENS"	4	10	5	(POW ABS_ENS)
"POW(NATURAL)"	57	33	15	(POW (POW INTEGER))
"INTER (cst, cst2).((cst : INT) & (cst2 : ABS_ENS) | ({cst}) *s ({cst2}))"	120	50	33	(POW (PRODUCT INTEGER ABS_ENS))
"0"	82	40	29	INTEGER
"inter_2"	15	17	36	(POW INTEGER)
"cst2"	19	13	10	BOOL
"{Elt1, Elt2}"	3	-1	-1	(POW ENS)
"{TRUE,FALSE}"	81	40	12	(POW BOOL)
"ens"	6	16	5	(POW (PRODUCT BOOL INTEGER))
"{cst}"	115	50	64	(POW INTEGER)
"POW(NAT1)"	54	32	14	(POW (POW INTEGER))
"ens2"	7	26	5	(POW INTEGER)
"ens4"	9	16	22	(POW INTEGER)
"cst"	18	13	5	ABS_ENS
"FIN(ens2)"	29	27	12	(POW (POW INTEGER))
"union_2"	14	17	27	(POW INTEGER)
"ens5"	10	16	28	(POW (PRODUCT (PRODUCT (POW ABS_ENS) (POW REAL)) (POW FLOAT)))
"NATURAL"	56	33	19	(POW INTEGER)
"FLOAT"	41	29	47	(POW FLOAT)
"UNION (cst, cst2).((cst : INT) & (cst2 : ABS_ENS) | ({cst}) *s ({cst2}))"	111	49	33	(POW (PRODUCT INTEGER ABS_ENS))
"1"	88	41	14	INTEGER
"POW(ABS_ENS)"	33	29	16	(POW (POW ABS_ENS))
"POW(NAT)"	51	31	14	(POW (POW INTEGER))
"NATURAL1"	59	34	19	(POW INTEGER)
"{ens, ens2, ens3 | (ens <: ABS_ENS) & (ens2 <: REAL) & (ens3 <: FLOAT)}"	78	39	13	(POW (PRODUCT (PRODUCT (POW ABS_ENS) (POW REAL)) (POW FLOAT)))
"POW(FLOAT)"	42	29	43	(POW (POW FLOAT))
"((POW(ABS_ENS)) *s (POW(REAL))) *s (POW(FLOAT))"	44	29	41	(POW (PRODUCT (PRODUCT (POW ABS_ENS) (POW REAL)) (POW FLOAT)))
"{cst2}"	116	50	72	(POW ABS_ENS)
"REAL"	35	29	35	(POW REAL)
"POW(((POW(ABS_ENS)) *s (POW(REAL))) *s (POW(FLOAT)))"	47	29	12	(POW (POW (PRODUCT (PRODUCT (POW ABS_ENS) (POW REAL)) (POW FLOAT))))
"cst"	113	50	23	INTEGER
"(BOOL) *s (INTEGER)"	22	25	20	(POW (PRODUCT BOOL INTEGER))
"POW(INTEGER)"	49	30	12	(POW (POW INTEGER))
"POW((INTEGER) *s (ABS_ENS))"	70	36	15	(POW (POW (PRODUCT INTEGER ABS_ENS)))
"union_"	12	17	11	(POW INTEGER)
"NAT"	50	31	18	(POW INTEGER)
"(POW(ABS_ENS)) *s (POW(REAL))"	38	29	29	(POW (PRODUCT (POW ABS_ENS) (POW REAL)))
"ens"	72	39	14	(POW ABS_ENS)
"NAT1"	53	32	18	(POW INTEGER)
"union({ens2,ens3})"	98	47	16	(POW INTEGER)
"POW(NATURAL1)"	60	34	15	(POW (POW INTEGER))
"(INTEGER) *s (ABS_ENS)"	62	35	27	(POW (PRODUCT INTEGER ABS_ENS))
"ens2"	73	39	19	(POW REAL)
"TRUE"	79	40	13	BOOL
"ens3"	74	39	25	(POW FLOAT)
"FALSE"	80	40	19	BOOL
"POW((INTEGER) *s (ABS_ENS))"	65	35	15	(POW (POW (PRODUCT INTEGER ABS_ENS)))
"10"	83	40	32	INTEGER
"(INTEGER) *s (ABS_ENS)"	67	36	27	(POW (PRODUCT INTEGER ABS_ENS))
"Elt1"	1	9	12	ENS
"(0) .. (10)"	84	40	30	(POW INTEGER)
"ens3"	8	16	16	(POW INTEGER)
"({TRUE,FALSE}) *s ((0) .. (10))"	85	40	26	(POW (PRODUCT BOOL INTEGER))
